### PR TITLE
Appellant Substitution | Show Alert on Source Appeal 

### PR DESCRIPTION
**Pull request by jcq**
Connects [CASEFLOW-1663](https://vajira.max.gov/browse/CASEFLOW-1663)

### Description
This updates the Case Details page (on source appeal) to display an alert in the "About the Veteran" box if a substitution has taken place

### Acceptance Criteria
- [x] Code compiles correctly
- [x] Alert shows in "About the Veteran" with working link to target/substitution appeal

### Testing Plan
1. Go to Case Details page for an appeal _that has had a substitution performed_ (the source appeal)
2. Check the "About the Veteran" section and confirm that `Alert` shows
3. Click on link in the `Alert` and confirm it links to the target/substitution appeal

### Updated Screenshot
Screen Shot 2021-06-25 at 11 54 15

### CODE REVIEW

File: app/models/serializers/work_queue/appellant_substitution_serializer.rb [hunk 1]
```diff
@@ -22,14 +22,15 @@ class WorkQueue::AppellantSubstitutionSerializer
     object.source_appeal&.claimant&.name
   end
 
+  attribute :target_appeal_uuid do |object|
+    object.target_appeal&.uuid
+  end
+
```
Comment (jcq) [inline app/models/serializers/work_queue/appellant_substitution_serializer.rb:28]: Needed this to be able to link to the new/target/substitution appeal stream

File: client/app/queue/CaseDetailsView.jsx [hunk 2]
```diff
@@ -145,6 +145,7 @@ export const CaseDetailsView = (props) => {
   const decisionHasDismissedDeathDisposition = (decisionIssue) =>
     decisionIssue.disposition === 'dismissed_death';
 
+  const hasSubstitution = appeal.substitutions?.length;
```
Comment (jcq) [inline client/app/queue/CaseDetailsView.jsx:148]: Only tangentially related to the rest of this PR, this is just an opportunity to (hopefully) make code a little clearer

File: client/app/queue/CaseDetailsView.jsx [hunk 3]
```diff
@@ -243,7 +244,7 @@ export const CaseDetailsView = (props) => {
             <AppellantDetail
               title="About the Appellant"
               appeal={appeal}
-              substitutionDate={appeal.appellantSubstitution && appeal.appellantSubstitution.substitution_date}
+              substitutionDate={appeal.appellantSubstitution?.substitution_date} // eslint-disable-line camelcase
```
Comment (jcq) [inline client/app/queue/CaseDetailsView.jsx:247]: Code cleanup, unrelated to this ticket

File: client/app/queue/VeteranDetail.jsx [hunk 4]
```diff
@@ -74,15 +75,23 @@ const VeteranDetail = ({ veteran, stateOnly }) => {
...
-      <BareList ListElementComponent="ul" items={details.map(getDetailField)} />
-      <p><em>{COPY.CASE_DETAILS_VETERAN_ADDRESS_SOURCE}</em></p>
-    </ul>
+    <>
+      <div {...detailListStyling}>
+        <BareList ListElementComponent="ul" items={details.map(getDetailField)} />
+        <p><em>{COPY.CASE_DETAILS_VETERAN_ADDRESS_SOURCE}</em></p>
+        {substitutionAppealId && <AppealHasSubstitutionAlert targetAppealId={substitutionAppealId} />}
+      </div>
+    </>
```
Comment (jcq) [inline client/app/queue/VeteranDetail.jsx:84]: The previous syntax ended up with invalid child elements for the outer `ul`; this switches to a `div` to be better semantically correct.

Only other addition is the conditional display of `AppealHasSubstitutionAlert`

File: client/app/queue/VeteranDetail.stories.js [hunk 5]
```diff
@@ -7,21 +8,28 @@ import { amaAppeal, veteranInfo } from '../../test/data/appeals';
 export default {
   title: 'Queue/VeteranDetail',
   component: VeteranDetail,
+  decorators: [
+    (Story) => (
+      <MemoryRouter>
+        <Story />
+      </MemoryRouter>
+    ),
+  ],
```
Comment (jcq) [inline client/app/queue/VeteranDetail.stories.js:17]: Necessary now due to the inclusion of possible `Link` component (if alert is showing)

File: client/app/queue/VeteranDetail.stories.js [hunk 6]
```diff
@@ -37,3 +45,10 @@ Loading.args = { loading: true, veteranInfo: null };
 
 export const Error = Template.bind({});
 Error.args = { error: true, veteranInfo: null };
+
+export const WithSubstitution = Template.bind({});
+WithSubstitution.args = {
+  substitutions: [{
+    target_appeal_uuid: 'abc123',
```
Comment (jcq) [inline client/app/queue/VeteranDetail.stories.js:52]: This is not a UUID, but it is easier for a human to read.

File: client/app/queue/substituteAppellant/caseDetails/AppealHasSubstitutionAlert.jsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+export const AppealHasSubstitutionAlert = ({ targetAppealId }) => {
+  return (
+    <Alert type="info">
+      {SUBSTITUTE_APPELLANT_SOURCE_APPEAL_ALERT_DESCRIPTION}{' '}
+      <span>
+        See the new{' '}
+        <Link to={`/queue/appeals/${targetAppealId}`}>appeal stream</Link>.
+      </span>
+    </Alert>
```
Comment (jcq) [inline client/app/queue/substituteAppellant/caseDetails/AppealHasSubstitutionAlert.jsx:16]: Considered having the link be part of Markdown-formatted entry in `COPY.json`, but that would have precluded just doing a React Router link; this seemed easiest way to not necessitate a full-page reload if not needed.

File: client/app/queue/CaseDetailsView.jsx (hunk 3 above)
Comment (eileen-nava) [reply client/app/queue/CaseDetailsView.jsx:247]: I hope eslint start to recognize this syntax soon.

File: client/app/queue/VeteranDetail.jsx (hunk 4 above)
Comment (eileen-nava) [reply client/app/queue/VeteranDetail.jsx:84]: 🎉

**Review by eileen-nava — APPROVED** [review]
This looks good to me!  I didn't review it in storybook, but I tested it locally in the UI, so I think that's okay.

**Pull request merged**